### PR TITLE
Creating remaining classes and Create the associations between them

### PR DESCRIPTION
**Pull request by bienvenuushindi**
# Requirements :recycle: 
- [x] Create a class Classroom
- [x] Create the has-many/belongs-to relationship between Classroom and Student
- [x] Create a class Book
- [x] Create a class Rental
- [x] Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental

### CODE REVIEW

File: association/classroom.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  attr_accessor :label
+
+  def initialize(label)
+    @label = label
+    @students = []
+  end
+
+  def add_student(student)
+    @students < student
+  end
```
Comment (Munsa1) [inline association/classroom.rb]: - [ ] Kindly implement the method that serves as the 'has-many' (a classroom has many students). like the foloowing example
image

File: classes/student.rb [hunk 2]
```diff
@@ -1,4 +1,6 @@
 class Student < Person
+  attr_reader :classroom
+
   def initialize(age, classroom, name = 'Unknown', parent_permission: true)
     @classroom = classroom
```
Comment (Munsa1) [inline classes/student.rb:5]: - [ ] It is a requirement that When setting the classroom for a student it also adds it to the classrooms' students:
e.g
``classroom.students.push(self) unless classroom.students.include?(self)``

File: classes/person.rb [hunk 3]
```diff
@@ -8,13 +8,18 @@ def initialize(age, name = 'Unknown', parent_permission: true)
...
     super()
   end
 
   def correct_name
     @name
   end
 
+  def add_rental(rental)
+    @rentals < rental
+  end
```
Comment (Munsa1) [inline classes/person.rb]: - [ ] The add_rental method shpuld be implemented as follows, so as to achieve the has-many association.
image

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @bienvenuushindi, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
:heavy_check_mark: Descriptive Readme
:heavy_check_mark: Passing Linters
:heavy_check_mark: Correct Gitflow
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classes/person.rb (hunk 3 above)
Comment (Wes-Isaac) [inline classes/person.rb]: - [ ] Great job so far, in order to achieve the` has-many` association the person itself needs to be sent. Kindly fix that.

`i.e  Rental.new(....)`

**Review by Wes-Isaac — CHANGES_REQUESTED** [review]
Hi @bienvenuushindi, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Highlights
- No linter error ✅.
- Great PR description 👍.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by m05e5 — APPROVED** [review]
# Status: ✔️ Project Approved ✔️ 

Hello @bienvenuushindi   🔥 

Good job on implementing the requested changes. You smashed it. Well Done.
Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

goodjob


Now you can merge this branch 🚀 

Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

_m05e5_

### 💻 👩🏽‍💻  Happy Coding!  👨🏽‍💻 🖥️

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**